File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x:x[0])
        ref = points[0][1]
        result = 1

        for s, e in points[1:]:
            if s > ref:
                result += 1
                ref = e
            else:
                ref = min(ref, e)

        return result

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a, b) -> Integer.compare(a[0], b[0]));
        int ref = points[0][1];
        int result = 1;

        for (int i = 1; i < points.length; i++) {
            if (points[i][0] > ref) {
                result++;
                ref = points[i][1];
            } else {
                ref = Math.min(ref, points[i][1]);
            }
        }

        return result;
    }
}